Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -86,6 +86,81 @@
 TIME_IS_MISSING = 'time_is_missing'
 
 
+# =========== Enrich User Mechanism ============
+class UserMappingObject:
+    def __init__(self, service, should_map_user, table_name='splunk_xsoar_users', xsoar_user_column_name='xsoar_user',
+                 splunk_user_column_name='splunk_user'):
+        self.service = service
+        self.should_map = should_map_user
+        self.table_name = table_name
+        self.xsoar_user_column_name = xsoar_user_column_name
+        self.splunk_user_column_name = splunk_user_column_name
+
+    def _get_record(self, col, value_to_search):
+        """ Gets the records with the value found in the relevant column. """
+        return self.service.kvstore[self.table_name].data.query(**{"query": json.dumps({col: value_to_search})})
+
+    def get_xsoar_user_by_splunk(self, splunk_user):
+
+        record = self._get_record(self.splunk_user_column_name, splunk_user)
+
+        if not record:
+            demisto.error(
+                "Could not find xsoar user matching splunk's {0}. Consider adding it to the {1} lookup.".format(
+                    splunk_user, self.table_name))
```

Comment:
When using multiple keys in `format` it's best practice to provide the actual key, i.e.
```suggestion
            demisto.error(
                "Could not find xsoar user matching splunk's {splunk_user}. Consider adding it to the {table_name} lookup.".format(
                    splunk_user=splunk_user, table_name=self.table_name))
```